### DISCUSSION TITLE
I want to alert when there's no data for some time period, what should I set "If no data or all values are null" to?

### DISCUSSION THREAD

**Discussion by in-op** [grafana/grafana#45695] [Alerting]
My datasource will periodically not send data for a few tens of minutes at a time, so I want to alert only if there has been a full hour of no data. I've set the alert condition to `WHEN last() OF query(A, 1h, now) HAS NO VALUE`. I'm unsure which option I should select for `If no data or all values are null`? It seems like each option overrides what I want to happen:

- Alerting (I don't want to alert immediately)
- No Data (Same as alerting, just different text)
- Keep Last State (does this have priority over the condition, and will erroneously keep it in the Ok state?)
- Ok (same, does this keep it in the Ok state?)

**Comment by thestraycat** [comment]
Did you work this out? I have the exact same problem! I'd love to hear how you might have tackled it...

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
